framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -44,24 +46,29 @@ AutomaticMortarGeneration::AutomaticMortarGeneration(
     bool periodic)
   : ConsoleStreamInterface(app),
     mesh(mesh_in),
-    mortar_segment_mesh(mesh_in.comm()),
     h_max(0.),
     _debug(false),
     _on_displaced(on_displaced),
-    _periodic(periodic)
+    _periodic(periodic),
+    _distributed(!mesh.is_replicated())
 {
   primary_secondary_boundary_id_pairs.push_back(boundary_key);
   primary_requested_boundary_ids.insert(boundary_key.first);
   secondary_requested_boundary_ids.insert(boundary_key.second);
   primary_secondary_subdomain_id_pairs.push_back(subdomain_key);
   primary_boundary_subdomain_ids.insert(subdomain_key.first);
   secondary_boundary_subdomain_ids.insert(subdomain_key.second);
+
+  if (_distributed)
+    mortar_segment_mesh = libmesh_make_unique<DistributedMesh>(mesh.comm());
+  else
+    mortar_segment_mesh = libmesh_make_unique<ReplicatedMesh>(mesh.comm());
```

Comment:
`_distributed` should simply derive from the parent  mesh. For example, if I am using 3D distributed mesh, I would expect 2D mortar mesh is distributed as well.

Reply:
That's why I'm doing...right?